server/models.py

```diff
@@ -0,0 +1,72 @@
+from app import db
+from sqlalchemy.sql import func
+from sqlalchemy import ForeignKey
+
+class contest_user(db.Model):
+    __tablename__ = 'contest_user'
+
+    id = db.Column(db.Integer, primary_key=True)
+    email = db.Column(db.String())
+    password = db.Column(db.String())
+    createdTime = db.Column(db.DateTime, default=func.utcnow())
+
+    def __init__(self, id, email, password, createdTime):
+        self.id = id
+        self.email = email
+        self.password = password
+        self.createdTime = createdTime
+
+class submission_user(db.Model):
+    __tablename__ = 'submission_user'
+
+    id = db.Column(db.Integer, primary_key=True)
+    sub_email = db.Column(db.String())
+    sub_password = db.Column(db.String())
+    createdTime = db.Column(db.DateTime, default=func.utcnow())
+
+class contest_table(db.Model):
+    __tablename__ = 'contest'
+
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String())
+    description = db.Column(db.String())
+    prizeContest = db.Column(db.Integer())
+    deadlineDate = db.Column(db.DateTime)
+    createdTime = db.Column(db.DateTime, default=func.utcnow())
+    updateTime = db.Column(db.DateTime, onupdate=func.utcnow()) # store updated time of row
+    contestCreater = db.Column(db.Integer, ForeignKey('contest_user.id'))
+
+    def __init__(self, id, title, description, prizeContest, deadlineDate, createdTime, updateTime, contestCreater):
+        self.id = id
```

Comment:
I don't think "id" should be one of the parameters when initializing a new object because isn't that auto generated?
Or am I misunderstanding this?

Reply:
you're right, this should be required to init a model - it should be auto generated and handled via Postgres